app_integrations/apps/aliyun.py

```diff
@@ -0,0 +1,148 @@
+
+from app_integrations import LOGGER
+from app_integrations.apps.app_base import StreamAlertApp, AppIntegration
+
+@StreamAlertApp
+class AliyunApp(AppIntegration):
+    """An app that collects events from the ActionTrail feature of Aliyun.
+
+    Results are paginated, with a NextToken field provided that is used in subsequent queries.
+    The result of a successful api call is json whose outermost schema is
+
+    {
+      "EndTime": <end of the time range of events>,
+      "NextToken": <the token to use to request the next page of data>,
+      "RequestId": <the ID of the request>,
+      "StartTime": <start of the time range of events>,
+      "Events": [entries],
+    }
+
+    If there are no more events in the queried range, the 'NextToken' element is not present.
+
+    Resource:
+        https://www.alibabacloud.com/help/doc-detail/28849.htm
+    """
+
+    _MAX_RESULTS = 50
+
+    def __init__(self, config):
+        super(AliyunApp, self).__init__(config)
+        self._next_token = None
+
+    @classmethod
+    def _type(cls):
+        return 'events'
+
+    @classmethod
+    def service(cls):
+        return 'aliyun'
+
+
+    @classmethod
+    def date_formatter(cls):
+        """Return a format string for a date, ie: 2014-05-26T12:00:00Z
+
+        This format is consisten with the format used by the Aliyun API:
+            https://www.alibabacloud.com/help/doc-detail/28849.htm
+        """
+        return '%Y-%m-%dT%H:%M:%SZ'
+
+    def _gather_logs(self):
+        auth = self._config.auth
+        client = AcsClient(auth['access_key_id'], auth['access_key_secret'], auth['region_id'])
+
+        request = LookupEventsRequest.LookupEventsRequest()
+        request.set_MaxResults(self._MAX_RESULTS)
+        request.set_StartTime = self._last_timestamp
+        if self._next_token:
+            request.set_NextToken(self._next_token)
+
+        try:
+            response = client.do_action_with_exception(request)
+            json_response = json.loads(response)
+            if 'NextToken' in json_response:
+                self._more_to_poll = True
+                self._next_token = json_response['NextToken']
+            else:
+                self._more_to_poll = False
+                self._last_timestamp = json_response['EndTime']
+
+            return json_response['Events']
+
+        except ServerException as e:
+            LOGGER.exception("Server Exception: %s", str(e))
+            return False
+        except ClientException as e:
+            LOGGER.exception("Client Exception: %s", str(e))
+            return False
+
+    @classmethod
+    def _required_auth_info(cls):
+        """Required credentials for access to the resources"""
+
+        def region_validator(region):
+            """Region names pulled from https://www.alibabacloud.com/help/doc-detail/40654.htm"""
+
+            if region in ["cn-qingdao", "cn-beijing", "cn-zhangjiakou", "cn-huhehaote",
```

Comment:
please use single (`'`) quotes on string literals and not double (`"`) quotes

Reply:
Small request, can we use `set` instead of `list`, it is not big a deal though.